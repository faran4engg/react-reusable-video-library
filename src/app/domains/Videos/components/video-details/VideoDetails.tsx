import { BadgePills } from 'app/domains/Common/components/badge-pills';
import { FC } from 'react';
import VideoViewMeta from '../video-view-meta/VideoViewMeta';
import { VideoDetailsProps } from './types';
import { useHistory } from 'react-router-dom';

const VideoDetails: FC<VideoDetailsProps> = ({
  details: { title, description, category, speakerName, tags },
  watchVideo,
  showMoreDetails,
  videoViewUrl,
  showWatchBtn = true,
}) => {
  const history = useHistory();

  return (
    <div className="flex flex-col items-center justify-center capitalize cursor-default ">
      <div className="w-full overflow-hidden shadow-lg bg-light-mode dark:bg-dark-mode">
        <h3 className="px-3 py-1 font-semibold tracking-wide text-center truncate">
          {title}
        </h3>
        {showMoreDetails && (
          <>
            <p className="px-3 py-1 my-1 text-xs tracking-wide text-left line-clamp-2 hover:line-clamp-none">
              📝 {description}
            </p>
            <div className="flex items-center justify-between px-3 py-1 pb-2 ">
              <span
                className="px-2 py-1 text-xs font-semibold bg-gray-200 rounded cursor-pointer dark:bg-gray-900"
                onClick={() => history.push(`/search/categories/${category}`)}
              >
                📌 &nbsp; {category}
              </span>

              <span
                className="px-2 py-1 text-xs font-semibold bg-gray-200 rounded cursor-pointer dark:bg-gray-900"
                onClick={() =>
                  history.push(`/search/speakerNames/${speakerName}`)
                }
              >
                🔊 &nbsp; {speakerName}
              </span>
            </div>
            <div className="px-3 py-1 text-center">
              {tags.map(tag => (
                <BadgePills
                  key={tag}
                  name={tag}
                  onClick={() => history.push(`/search/tags/${tag}`)}
                />
              ))}
            </div>
          </>
        )}
        <VideoViewMeta
          watchVideo={watchVideo}
          showWatchBtn={showWatchBtn}
          videoViewUrl={videoViewUrl}
          title={title}
        />
      </div>
    </div>
  );
};

export default VideoDetails;
