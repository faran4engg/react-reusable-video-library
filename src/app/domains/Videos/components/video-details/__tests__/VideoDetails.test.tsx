import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import VideoDetails from '../VideoDetails';

const renderer = createRenderer();

describe('<VideoDetails />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(
      <VideoDetails
        details={{
          title: 'title',
          description: 'description',
          category: 'category',
          speakerName: 'NAK',
          tags: ['tagsA', 'tagB'],
        }}
        showMoreDetails
        videoViewUrl="/path/id"
      />,
    );
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
