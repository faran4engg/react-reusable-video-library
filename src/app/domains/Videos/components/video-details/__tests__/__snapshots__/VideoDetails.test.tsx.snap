// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoDetails /> should render and match the snapshot 1`] = `
<div
  className="flex flex-col items-center justify-center capitalize cursor-default "
>
  <div
    className="w-full overflow-hidden shadow-lg bg-light-mode dark:bg-dark-mode"
  >
    <h3
      className="px-3 py-1 font-semibold tracking-wide text-center truncate"
    >
      title
    </h3>
    <React.Fragment>
      <p
        className="px-3 py-1 my-1 text-xs tracking-wide text-left line-clamp-2 hover:line-clamp-none"
      >
        📝 
        description
      </p>
      <div
        className="flex items-center justify-between px-3 py-1 pb-2 "
      >
        <span
          className="px-2 py-1 text-xs font-semibold bg-gray-200 rounded cursor-pointer dark:bg-gray-900"
          onClick={[Function]}
        >
          📌   
          category
        </span>
        <span
          className="px-2 py-1 text-xs font-semibold bg-gray-200 rounded cursor-pointer dark:bg-gray-900"
          onClick={[Function]}
        >
          🔊   
          NAK
        </span>
      </div>
      <div
        className="px-3 py-1 text-center"
      >
        <BadgePills
          name="tagsA"
          onClick={[Function]}
        />
        <BadgePills
          name="tagB"
          onClick={[Function]}
        />
      </div>
    </React.Fragment>
    <Memo(VideoViewMeta)
      showWatchBtn={true}
      title="title"
      videoViewUrl="/path/id"
    />
  </div>
</div>
`;
